packages/amplify-graphql-index-transformer/src/utils.ts

```diff
@@ -50,3 +51,60 @@ export function validateNotSelfReferencing(config: IndexDirectiveConfiguration |
     }
   }
 }
+
+/**
+ * Checks if @auth owner field has been set to a sortKeyField for @primaryKey
+ */
+export const validateNotOwnerAuth = (
+  sortKeyField: string,
+  { object }: PrimaryKeyDirectiveConfiguration,
+  ctx: TransformerContextProvider,
+): boolean => {
+  const authDir = object.directives?.find(dir => dir.name.value === 'auth');
+
+  if (!authDir) return true;
+
+  const dirRules = (authDir.arguments?.find(arg => arg.name.value === 'rules')?.value as ListValueNode)?.values || [];
+  const hasOwnerFieldAsSortKey = dirRules.some(rule => {
+    let isOwner = false;
+    let identityClaimIsSet = false;
+    let ownerFieldIsSet = false;
+    let usesMultiClaim = false;
+    let sortKeyFieldIsAuthField = false;
+
+    (rule as ObjectValueNode).fields?.forEach(field => {
+      const name: string = field.name.value;
+      const { value } = field?.value as StringValueNode;
+
+      if (name === 'allow' && value === 'owner') {
+        isOwner = true;
+      }
+
+      if (name === 'identityClaim') {
+        identityClaimIsSet = true;
+
+        if (value === 'sub::username') {
+          usesMultiClaim = true;
+        }
+      }
+
+      if (name === 'ownerField') {
+        ownerFieldIsSet = true;
+
+        if (value === sortKeyField) {
+          sortKeyFieldIsAuthField = true;
+        }
+      }
+    });
+
+    const featureFlagEnabled = ctx.featureFlags.getBoolean('useSubUsernameForDefaultIdentityClaim');
```

Comment:
If this value isn't set in the project, `.getBoolean` will return the default value as configured between "existing" and "new" Amplify apps. (https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-cli-core/src/feature-flags/featureFlags.ts#L218)

Reply:
(Also, the feature flags module always errors out if the feature flag is missing from the instance because it acts as a singleton, so it always needs to be set in tests too.)